src/modifiers.ts

```diff
@@ -92,190 +93,200 @@ function hasSpecialChars(str: string): boolean {
 
 export class Modifiers {
     codeModel: CodeModel;
-    allCommandGroups: string[];
+    allCommandGroups: Record<string, number>;
 
     constructor(protected session: Session<CodeModel>) {
         this.codeModel = session.model;
         this.getAllCommandGroup();
     }
 
     getAllCommandGroup() {
-        this.allCommandGroups = [];
-        for (const group of values(this.codeModel.operationGroups)) {
-            this.allCommandGroups.push(group.language['az'].command);
-        }
+        this.allCommandGroups = {};
+        let groupIdx = -1;
+        this.codeModel.operationGroups.forEach((operationGroup) => {
+            groupIdx++;
+            this.allCommandGroups[operationGroup.language['az'].command] = groupIdx;
+        });
     }
 
     async process() {
         directives = AzConfiguration.getValue(CodeGenConstants.directive);
         const options = AzConfiguration.getValue(CodeGenConstants.az);
-        if (!isNullOrUndefined(directives)) {
-            for (const directive of directives.filter((each) => !each.transform)) {
+        if (isNullOrUndefined(directives)) {
```

Comment:
Can we add return value for this function?

Reply:
Do we have test case for this function?

Reply:
seperate it into three functions operationGroupModify operationModify and paramModify

Reply:
added. we have unittest for this modifier plugin https://github.com/Azure/autorest.az/blob/616138f2d86d34cb6238e43359cf97738fa2f1bb/test/test-modifier.ts